manager/controllers/app/plotter_generator.go

```diff
@@ -201,6 +201,77 @@ func (p *PlotterGenerator) addStep(element *ResolvedEdge, datasetID string, api
 	return steps
 }
 
+// getSupportedFormat returns the first dataformat supported by the module's capability sink interface
+func (p *PlotterGenerator) getSupportedFormat(capability *app.ModuleCapability) taxonomy.DataFormat {
+	for _, inter := range capability.SupportedInterfaces {
+		if inter.Sink == nil {
```

Comment:
rewrite:
for ... {
    if inter.Sink != nil {
        return inter.Sink.DataFormat
    }
}    
...